testing/travis.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python2
+from __future__ import print_function
+
+import base64
+import json
+import logging
+import os
+import os.path
+import socket
+import ssl
+import subprocess
+import sys
+import urllib2
+
+# FROM plugins/inventory/terraform.py
+# PYTHON_PATH is added to in .travis.yml
+import terraform as tf
+
+## TRAVIS STEPS
+
+def install():
+    cmds = [
+        (["pip", "install", "-r", "requirements.txt"], 1),
+        (["curl", "-SL0", "terraform.zip",  "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip"], 1),
+        (["unzip", "-d", "testing/cache/", "terraform.zip"], 1),
+        (["ssh-keygen", "-N", '""', "-f", "~/.ssh/id_rsa"], 1),
+            ]
+
+    run_cmds(cmds)
+
+
+def script():
+
+    # Filter out commits that are documentation changes.
+    commit_range = os.environ.get("TRAVIS_COMMIT_RANGE", "master..HEAD")
+    diff_names = str(subprocess.check_output(["git",  "diff",  "--name-only", commit_range]))
+
+    not_docfiles = filter_not_docfiles(diff_names)
+    if len(not_docfiles) < 1:
+        logging.info("All of the changes were in documentation. Skipping build.")
+        sys.exit(0)
+
+    linting_providers = ["clc", "softlayer", "triton"]
+    deploy_providers = ["gce", "aws", "do"]
+
+    cmds = linter_cmds()
+
+    if os.environ.get("PROVIDER", None) in deploy_providers:
+        cmds.extend(deploy_to_cloud())
+
+    if not run_cmds(cmds, fail_sequential=True):
+        sys.exit(1)
+
+    health_checks()
```

Comment:
Calling it from the config file would map better with how it is currently implemented, also. I'll change it.